src/simulation/MixedLJSimulation.cpp

```diff
@@ -128,51 +136,82 @@ MixedLJSimulation::MixedLJSimulation(
     if (n_thermostat) {
         // Initialize thermostat
         spdlog::info(
-            "Initializing thermostat with T_init={}K, T_target={}K, delta_T={}K with a frequency "
-            "of {}",
+            "Initializing thermostat of type={} with T_init={}K, T_target={}K, delta_T={}K with"
+            " a frequency of {}",
+            thermostat->getName(),
             T_init,
             T_target,
             delta_T,
             n_thermostat);
-        thermostat = Thermostat(T_init, T_target, delta_T, cellGrid.gridDimensionality);
 
         // Intialize temperature
         spdlog::info("Setting initial temperature to {} K", T_init);
-        thermostat.initializeBrownianMotion(this->container);
-    }
-    else{
+        thermostat->initializeBrownianMotion(*this);
+    } else {
         spdlog::info("Themostat is turned off.");
     }
 }
 
 void MixedLJSimulation::runSim()
 {
+
+    if (container.particles.size() == 2500) {
+        container.particles[874].setType(2);
+        container.particles[875].setType(2);
+        container.particles[924].setType(2);
+        container.particles[925].setType(2);
+    }
+
+
     auto startTime = std::chrono::steady_clock::now();
     unsigned long long particleUpdates = 0;
     while (time < end_time) {
         bcHandler.preUpdateBoundaryHandling(*this);
 
+        spdlog::debug("Force calculation...");
         strategy.calF(*this);
+        spdlog::debug("Velocity calculation...");
+
+        if (time < 150 && container.particles.size() == 2500) {
+            std::array<double, 3> Fz_up = { 0, 0, 0.8 };
+            Particle& p1 = container.particles[874];
+            Particle& p2 = container.particles[875];
+            Particle& p3 = container.particles[924];
+            Particle& p4 = container.particles[925];
+            p1.setOldF(p1.getF());
+            p1.setF(p1.getF() + Fz_up);
```

Comment:
Yes, we apparently missed something.
The `setOldF` calls are not necessary and do change the math slightly. This would lead to the fact that the current and old forces are always equal except for the upward pulling force.

To fix this issue, the `setOldF` calls must be removed. And after time>150, the issue is resolved by simply not entering the if statement anymore.

Reply:
Yes, so I thought. Surprisingly the simulation still looks fine :D anyway, I resolve this discussion too :)